docs/reactive-nonblocking.qmd

```diff
@@ -0,0 +1,316 @@
+
+### Invoking an extended task
+
+Unlike a reactive effect, simply creating an extended task does not cause it to run.
+It needs to be invoked (called like a function).
+
+In this case, the `sum_values` extended task is called from the `btn_click` reactive effect (`@reactive.effect`), which runs whenever the button is clicked (`@reactive.event(input.btn)`).
+
+Notice also that the `sum_values` logic no longer reads `input.x()` and `input.y()` directly; as an extended task, it is not allowed to read reactive inputs or calcs (and attempting to do so would result in an error).
+Instead, it takes `x` and `y` as arguments, which are passed in by `btn_click`.
+
+### Using the result of an extended task
+
+The `sum` output is now a regular reactive calc, which reads `sum_values.result()` to get the result of the extended task.
+This output actually does not "wait for" the extended task to complete, exactly; instead, it will run multiple times, as the extended task goes through different states.
+For each state, `sum_values.result()` will behave differently:
+
+* **Not yet clicked:** Raises a silent exception, which will cause the `sum` output to display nothing.
+* **Running:** Raises a special type of exception that tells Shiny to keep the output in the "in progress" state.
+* **Successfully completed:** Returns the return value of `sum_values`, in this case an integer.
+* **Completed with an exception:** If `sum_values` raised an exception while processing, then re-raise that same exception, causing it to be displayed to the user in the `sum` output.
+
+It's not necessary to memorize these states.
+Just remember that `sum_values.result()` is a reactive, synchronous method that knows how to do right thing based on the state of the extended task.
+
+### Other features of extended tasks
+
+#### Cancel a running task
+
+Although it's not shown in the example above, you can also cancel an extended task by calling the `cancel()` method (for example, `sum_values.cancel()`).
+This will attempt to cancel the asyncio task that is running the extended task, and will cause `sum_values.result()` to raise a silent exception.
+
+Calling `sum_values.cancel()` on a task that isn't running will have no effect.
+
+#### Multiple invocations
+
+An extended task can run concurrently to reactive code and to other extended tasks--that's its whole purpose.
+However, a single extended task object cannot run itself multiple times concurrently. If you call `sum_values()` while it is already running, it will enqueue the new invocation and run it after the first one completes.
+
+This is often not the behavior you want, especially if the task takes a long time to complete.
+A user may accidentally click an action button twice, or they may click it again because they think the first click didn't work.
+To prevent this, use replace `ui.input_action_button` with `ui.input_task_button`, which will automatically prevent subsequent clicks until the task completes.
```

Comment:
```suggestion
To prevent this, use `ui.input_task_button` instead of `ui.input_action_button` to invoke the task since the former automatically prevents subsequent clicks until the task completes.
```